### PR TITLE
Add Guide For Spring On Kubernetes

### PR DESCRIPTION
**Pull request by ryanjbaxter**
This is a topical guide to help developers get started using Spring on Kubernetes.

### CODE REVIEW

File: app/src/main/java/com/example/demo/DemoApplication.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	public static void main(String[] args) {
+		SpringApplication.run(DemoApplication.class, args);
+	}
+
+	@GetMapping
+	public Mono<String> index() {
+		return webClient.get().uri("http://k8s-workshop-name-service").exchange().flatMap(clientResponse -> {
+			String host = clientResponse.headers().asHttpHeaders().get("k8s-host").get(0);
+			return clientResponse.bodyToMono(String.class).flatMap(name -> Mono.just("Hello " + name + " from " + host));
+		});
```
Comment (rstoyanchev) [inline app/src/main/java/com/example/demo/DemoApplication.java]:
```suggestion
		return webClient.get().uri("http://k8s-workshop-name-service")
				.retrieve()
				.toEntity(String.class)
				.map(entity -> {
					String host = entity.getHeaders().get("k8s-host").get(0);
					return "Hello " + entity.getBody() + " from " + host;
				});

```

Happened to notice this.. The `exchange()` method is difficult to use correctly, most often it is not necessary, and is also deprecated in Spring Framework 5.3.

File: app/src/main/java/com/example/demo/DemoApplication.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@SpringBootApplication
+@RestController
+public class DemoApplication {
+
+	private WebClient webClient = WebClient.builder().build();
```
Comment (rstoyanchev) [inline app/src/main/java/com/example/demo/DemoApplication.java]:
```suggestion
	private WebClient webClient = WebClient.create();
```

**Review by rstoyanchev — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demo/DemoApplication.java (hunk 1 above)
Comment (ryanjbaxter) [reply app/src/main/java/com/example/demo/DemoApplication.java]: Thanks @rstoyanchev!

File: README.adoc [hunk 3]
```diff
@@ -1,91 +1,511 @@
...
+
+Kubernetes sets up DNS entries so that we can use the service ID for the `name-service` to make an HTTP request to the
+service without knowing the IP address of the pods. The Kubernetes service also load balances these requests between
+all the pods.
+
+In your application, open `DemoApplication.java` in `src/main/java/com/example/demo`. Modify the code as follows:
+
+[source,java]
+----
+include::app/src/main/java/com/example/demo/DemoApplication.java[]
```
Comment (Haybu) [inline README.adoc:451]: not rendering source code

**Review by Haybu — APPROVED** [review]

File: README.adoc (hunk 3 above)
Comment (ryanjbaxter) [reply README.adoc:451]: Yeah not on GitHub but I think it will once everything gets on Sagan.  It works fine in Intellij

**Comment by gregturn** [comment]
Resolved via https://github.com/spring-guides/top-spring-on-kubernetes.

**Pull request closed without merging**